src/main/java/org/schemaspy/mutation/ImpliedConstraint.java

```diff
@@ -0,0 +1,121 @@
+package org.schemaspy.mutation;
+
+import org.schemaspy.model.DatabaseObject;
+import org.schemaspy.model.ImpliedForeignKeyConstraint;
+import org.schemaspy.model.Table;
+import org.schemaspy.model.TableColumn;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Search supplied Collection of tables and add ImpliedForeignKeyConstraints
+ * when a childColumn doesn't have a parent and finds a primary key that
+ * matches based on name and type.
+ */
+
+public class ImpliedConstraint implements Mutation<ImpliedForeignKeyConstraint> {
+
```

Comment:
No constructor means that nothing is encapsulated i.e., [the objects are not unique](https://www.yegor256.com/2014/11/20/seven-virtues-of-good-object.html#3-he-is-unique).

Reply:
It's more or less extraction of a static method.
So there was no encapsulation and no encapsulation has been added.